### ISSUE TITLE
Missing Oauth2 profile

### ISSUE THREAD

**Issue by adnansenyurt** [jhipster/jhipster-registry#221]
##### **Overview of the issue**

Setup requires Oauth2 profile which does not exist in project repository

##### **Motivation for or Use Case**

Connect Jhipster-Registry to a Keycloak server on cloud foundry (not local)

##### **Reproduce the error**

./mvnw package -DskipTests=true -B -Pprod,oauth2

results in:

[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:02 min
[INFO] Finished at: 2017-12-28T13:41:24+03:00
[INFO] Final Memory: 66M/454M
[INFO] ------------------------------------------------------------------------
**[WARNING] The requested profile "oauth2" could not be activated because it does not exist.**

Keycloak login page does not display when jhipster-registry is launched.

Note:

##### **Related issues**

Issue #148

##### **Suggest a Fix**

- Oauth2 profile should be defined in the repository
- A clear guidance on how to deploy micro services to cloud foundry with OIDC option would be helpful. Existing documentation is for localhost deployment.

##### **JHipster Registry Version(s)**
4.
##### **Browsers and Operating System**

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by jdubois** [comment]
Could you point to where you saw in the documentation that this profile should be used?

**Comment by adnansenyurt** [comment]
There is no documentation on JHipster site for deploying jhipster-registry with Keycloak to cloud foundry. All I could fing was your comment on the closed issue of JHipster:
https://github.com/jhipster/jhipster-registry/issues/193#issuecomment-338006563

**Comment by jdubois** [comment]
Oh that comment was done while we were developing the integration, I don't think this profile has been released at all, it's not needed.
Did you read http://www.jhipster.tech/security/ ? Basically you need to set up those variables with CloudFoundry - that's what the Docker Compose configuration does automatically.

**Comment by adnansenyurt** [comment]
I've read it before for micro service deployment but did not consider these for jhipster-registry app. My mistake.

So, application-oauth2.yml is not used. 

- I will move OIDC settings there to application.yml? or to application-cloudfoundry.xml?
- What about the OIDC URL settings? Shall I change all the 'http://localhost:9080' strings to keycloak external URI on cloud?
- When deploying Keycloak with docker to cloud foundry, shall I also modify the http://localhost:9080 settings in jhipster-realm.json file?







# ===================================================================
# OpenID Connect Settings. Default settings are for Keycloak in Docker.
# Start Keycloak using `docker-compose -f src/main/docker/keycloak.yml up`.
#
# To use Okta, your settings should resemble the following:
#
# accessTokenUri: https://{yourOktaDomain}.com/oauth2/default/v1/token
# userAuthorizationUri: https://{yourOktaDomain}.com/oauth2/default/v1/authorize
# clientId: {copy after creating an OIDC app}
# clientSecret: {copy after creating an OIDC app}
# userInfoUri: https://{yourOktaDomain}.com/oauth2/default/v1/userinfo
# tokenInfoUri: https://{yourOktaDomain}.com/oauth2/default/v1/introspect
#
# You can also override these properties using environment variables.
# For example:
#
# export SECURITY_OAUTH2_CLIENT_CLIENT_ID="jhipster"
# export SECURITY_OAUTH2_CLIENT_CLIENT_SECRET="38561a05"
# ===================================================================
security:
    basic:
        enabled: false
    oauth2:
        client:
            access-token-uri: http://localhost:9080/auth/realms/jhipster/protocol/openid-connect/token
            user-authorization-uri: http://localhost:9080/auth/realms/jhipster/protocol/openid-connect/auth
            client-id: jhipster-registry
            client-secret: jhipster-registry
            client-authentication-scheme: form
            scope: openid profile email
        resource:
            user-info-uri: http://localhost:9080/auth/realms/jhipster/protocol/openid-connect/userinfo
            token-info-uri: http://localhost:9080/auth/realms/jhipster/protocol/openid-connect/token/introspect
            prefer-token-info: false
            jwt:
                key-uri: http://localhost:9080/auth/realms/jhipster

**Comment by jdubois** [comment]
Hold on. I'll have look this afternoon.

**Comment by pascalgrimaud** [comment]
1) you can package the app, only with the existing profiles in pom.xml :
- prod exists
- oauth2 doesn't exist
2) but you can start the app, using oauth2 profile. It will use application-oauth2.yml to load configuration

**Comment by jdubois** [comment]
Oh yes, thanks @pascalgrimaud !!

**Comment by pascalgrimaud** [comment]
Just tested and everything works like expected.
There is no bug here. In fact, `oauth2` is a profile for configuration, not a profile for compilation.

**Comment by adnansenyurt** [comment]
I've modified the bootstrap.xml as below.
----------
spring:
    application:
        name: jhipster-registry
    profiles:
        active: dev,native,oauth2
----------
This way I've managed to deploy & run locally, as you've suggested with the runtime oauth2 profile option. Thanks for your help.

I'm still struggling with deploying to cloud foundry and running on cloud. It seems There is an issue with deploying jhipster realm on cloud. I'll open a new issue to better describe.
